packages/api/src/beacon/routes/node.ts

```diff
@@ -77,148 +109,198 @@ export type NodeHealthOptions = {
+          metadata: ssz.altair.Metadata.fromJson(data.metadata),
+        }),
+        serialize: () => {
+          throw new Error("Not implemented");
+        },
+        deserialize: () => {
+          throw new Error("Not implemented");
+        },
+      },
+      meta: EmptyMetaCodec,
+    },
+  },
+  getPeers: {
+    url: "/eth/v1/node/peers",
+    method: "GET",
+    req: {
+      writeReq: ({state, direction}) => ({query: {state, direction}}),
+      parseReq: ({query}) => ({state: query.state, direction: query.direction}),
       schema: {query: {state: Schema.StringArray, direction: Schema.StringArray}},
     },
-    getPeer: {
-      writeReq: (peer_id) => ({params: {peer_id}}),
-      parseReq: ({params}) => [params.peer_id],
-      schema: {params: {peer_id: Schema.StringRequired}},
+    resp: {
+      ...JsonOnlyResponseCodec,
+      meta: {
+        toJson: (d) => d,
+        fromJson: (d) => ({count: (d as PeersMeta).count}),
+        toHeadersObject: () => ({}),
+        fromHeaders: () => ({}) as PeersMeta,
+      },
+      transform: {
+        toResponse: (data, meta) => ({data, meta}),
+        fromResponse: (resp) => resp as {data: NodePeer[]; meta: PeersMeta},
+      },
     },
-
-    getPeerCount: reqEmpty,
-    getNodeVersion: reqEmpty,
-    getSyncingStatus: reqEmpty,
-    getHealth: {
-      writeReq: (options) => ({
-        query: options?.syncingStatus !== undefined ? {syncing_status: options.syncingStatus} : {},
-      }),
-      parseReq: ({query}) => [{syncingStatus: query.syncing_status}],
-      schema: {query: {syncing_status: Schema.Uint}},
+  },
+  getPeer: {
+    url: "/eth/v1/node/peers/{peer_id}",
+    method: "GET",
+    req: {
+      writeReq: ({peerId}) => ({params: {peer_id: peerId}}),
+      parseReq: ({params}) => ({peerId: params.peer_id}),
+      schema: {params: {peer_id: Schema.StringRequired}},
     },
-  };
-}
-
-export function getReturnTypes(): ReturnTypes<Api> {
-  const stringType = new StringType();
-  const NetworkIdentity = new ContainerType(
-    {
-      peerId: stringType,
-      enr: stringType,
-      p2pAddresses: ArrayOf(stringType),
-      discoveryAddresses: ArrayOf(stringType),
-      metadata: ssz.altair.Metadata,
+    resp: JsonOnlyResponseCodec,
+  },
+  getPeerCount: {
+    url: "/eth/v1/node/peer_count",
+    method: "GET",
+    req: EmptyRequestCodec,
+    resp: {
+      data: PeerCountType,
+      meta: EmptyMetaCodec,
     },
-    {jsonCase: "eth2"}
-  );
-
-  const PeerCount = new ContainerType(
-    {
-      disconnected: ssz.UintNum64,
-      connecting: ssz.UintNum64,
-      connected: ssz.UintNum64,
-      disconnecting: ssz.UintNum64,
+  },
+  getNodeVersion: {
+    url: "/eth/v1/node/version",
+    method: "GET",
+    req: EmptyRequestCodec,
+    resp: JsonOnlyResponseCodec,
+  },
+  getSyncingStatus: {
+    url: "/eth/v1/node/syncing",
+    method: "GET",
+    req: EmptyRequestCodec,
+    resp: JsonOnlyResponseCodec,
```

Comment:
When we say our API have full SSZ support then it we should not have `JsonOnlyResponseCodec` expect the `lodestar` namespace.

Reply:
> When we say our API have full SSZ support 

This is not the claim of this PR, but it provides first-class support for ssz on the api. What we can claim though is that we have full ssz support on all performance relevant apis, that includes mostly beacon and validator required apis.

> If we require ALL endpoints to support SSZ before landing this feature, then we will likely never land this feature.

I don't even think this is practically useful. A lot of apis are not performance sensitive or mostly consumed by clients that do not even supports ssz, think dappnode querying the peer count.

I am not saying we shouldn't push for more apis to support ssz but there are limitations to it's usefulness in most of those cases.